packages/ui/src/pages/Register/IdentifierRegisterForm/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { SignInIdentifier } from '@logto/schemas';
+import classNames from 'classnames';
+import { useState, useCallback } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import Button from '@/components/Button';
+import ErrorMessage from '@/components/ErrorMessage';
+import type { IdentifierInputType } from '@/components/InputFields';
+import { SmartInputField } from '@/components/InputFields';
+import TermsOfUse from '@/containers/TermsOfUse';
+import useTerms from '@/hooks/use-terms';
+import { getGeneralIdentifierErrorMessage, validateIdentifierField } from '@/utils/form';
+
+import * as styles from './index.module.scss';
+import useOnSubmit from './use-on-submit';
+
+type Props = {
+  className?: string;
+  // eslint-disable-next-line react/boolean-prop-naming
+  autoFocus?: boolean;
+  signUpMethods: SignInIdentifier[];
+};
+
+type FormState = {
+  identifier: string;
+};
+
+const IdentifierRegisterForm = ({ className, autoFocus, signUpMethods }: Props) => {
+  const { t } = useTranslation();
+  const { termsValidation } = useTerms();
+  const [inputType, setInputType] = useState<IdentifierInputType>(
+    signUpMethods[0] ?? SignInIdentifier.Username
+  );
+
+  const { errorMessage, clearErrorMessage, onSubmit } = useOnSubmit();
+
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors, isSubmitted },
+  } = useForm<FormState>({
+    reValidateMode: 'onChange',
+    defaultValues: { identifier: '' },
+  });
+
+  const onSubmitHandler = useCallback(
+    async (event?: React.FormEvent<HTMLFormElement>) => {
+      clearErrorMessage();
+
+      void handleSubmit(async ({ identifier }, event) => {
+        event?.preventDefault();
```

Comment:
nit: it seems that `handleSubmit` will do this.

Reply:
updated